GovUk.Frontend.Umbraco.ExampleApp/uSync/v9/ContentTypes/govukfileuploadsettings.config

```
@@ -1,97 +1,81 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <ContentType Key="d10b177e-98b9-42c3-a7f6-632c885c6bcf" Alias="govukFileUploadSettings" Level="2">
-	<Info>
-		<Name>File upload settings</Name>
-		<Icon>icon-cloud-upload color-black</Icon>
-		<Thumbnail>folder.png</Thumbnail>
-		<Description>Settings for a file upload component. Ask users to upload something only if it’s critical to the delivery of your service.</Description>
-		<AllowAtRoot>False</AllowAtRoot>
-		<IsListView>False</IsListView>
-		<Variations>Nothing</Variations>
-		<IsElement>true</IsElement>
-		<HistoryCleanup>
-			<PreventCleanup>False</PreventCleanup>
-			<KeepAllVersionsNewerThanDays></KeepAllVersionsNewerThanDays>
-			<KeepLatestVersionPerDayForDays></KeepLatestVersionPerDayForDays>
-		</HistoryCleanup>
-		<Folder>GOV.UK</Folder>
-		<Compositions>
-			<Composition Key="bf69e8d6-97a6-4b50-bec7-c5993e120339">govukCssClasses</Composition>
-			<Composition Key="37cc75ea-be5c-4107-b9b8-0ef34f4b7c5e">govukErrorMessagePrefix</Composition>
-			<Composition Key="d508995b-a958-4273-8ab3-50735220a1b2">govukGrid</Composition>
-			<Composition Key="d7016563-0b6f-4c3a-b75d-b0bbe66419fd">govukGridColumnClasses</Composition>
-			<Composition Key="2bc3c434-f3ff-495d-b360-1a005245602e">govukLabelIsPageHeading</Composition>
-			<Composition Key="23e1a03d-dc42-4318-9e17-d458dde2dba6">govukModelProperty</Composition>
-			<Composition Key="f2365c7e-08c3-4ef9-abbc-e337e34de272">govukValidationRequired</Composition>
-		</Compositions>
-		<DefaultTemplate></DefaultTemplate>
-		<AllowedTemplates />
-	</Info>
-	<Structure />
-	<GenericProperties>
-		<GenericProperty>
-			<Key>cac445e1-9d87-452e-94de-a70e43ff09ff</Key>
-			<Name>Allowed file types</Name>
-			<Alias>errorMessageAllowedFileExtensions</Alias>
```

You've removed the `errorMessageAllowedFileExtensions` property but `ApplicationBuilderExtensions.cs` still refers to it and the validator that uses it still exists, you've only marked it as obsolete.

I had a chat with Joel about whether this validator should replace his earlier effort, and we agreed that the validation you're putting in place is better. However, Joel's suggestion was to incorporate the file extension check into this validator as a first pass which can fail fast, before reading into a `MemoryStream` for a further check. I agree with this as it'll give faster feedback to the user and helps to guard against an attacker uploading potentially malicious files into our memory. 

I can see you had a similar idea and have added an overload for `IsMatch` that checks the extension, but the suggestion here is that you would fail the validator as soon as the extension test fails, without ever doing the binary check in that case. Plus if you're consuming the package this logic would be in the validator rather than an `IsMatch` overload.

---

Moved the extension validation to the `AllowedFileTypesAttribute`

The reason why `AllowedFileExtensionsAttribute` is marked as Obsolete is because I don't know where it is used and do not want to create a breaking change. Marking is Obsolete allows people to migrate in an orderly fashion and we can delete it in a few months or so when necessary.

---

Yes, I understand your thinking on the use of `[Obsolete]`. 

However, `ApplicationBuilderExtensions.cs` connects the obsolete `AllowedFileExtensionsAttribute`  to a property with the alias `errorMessageAllowedFileExtensions`, so by removing that property you've broken it.

In this case I would recommend removing it because it was only added in v6.0.0-alpha004 and we haven't published v6.0.0 yet. It's the nature of alphas that features come and go, and projects use them at their own risk.

---

Removed `[AllowedFileExtensions]`